packages/react-components/react-breadcrumb/src/components/BreadcrumbButton/useBreadcrumbButton.ts

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+import type { BreadcrumbButtonProps, BreadcrumbButtonState } from './BreadcrumbButton.types';
+import { useButton_unstable, ButtonProps } from '@fluentui/react-button';
+import { useBreadcrumbContext_unstable } from '../Breadcrumb/BreadcrumbContext';
+
+/**
+ * Create the state required to render BreadcrumbButton.
+ *
+ * The returned state can be modified with hooks such as useBreadcrumbButtonStyles_unstable,
+ * before being passed to renderBreadcrumbButton_unstable.
+ *
+ * @param props - props from this instance of BreadcrumbButton
+ * @param ref - reference to root HTMLElement of BreadcrumbButton
+ */
+export const useBreadcrumbButton_unstable = (
+  props: BreadcrumbButtonProps,
+  ref: React.Ref<HTMLButtonElement | HTMLAnchorElement>,
+): BreadcrumbButtonState => {
+  const { appearance, iconPosition, size } = useBreadcrumbContext_unstable();
+
+  const { current = false, ...rest } = props;
+  const newProps = {
+    ...rest,
+    appearance: (props.appearance || appearance) as ButtonProps['appearance'],
+    iconOnly: Boolean(!props.children),
```

Comment:
```suggestion
```

This is probably redundant as `useButton` will override it:

<img width="541" alt="image" src="https://user-images.githubusercontent.com/14183168/231415148-b37a7cbe-bd5b-4a8a-a30f-2cd85b064702.png">